plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -151,40 +152,64 @@ fu_plugin_thunderbolt_rescan (FuPlugin *plugin, GError **error)
 		info->id = g_strdup (tdbid);
 		g_ptr_array_add (data->infos, info);
 
-		/* get the vendor ID */
-		rc = tbt_fwu_Controller_getVendorID (data->controllers[i],
-						     &info->vendor_id);
+		rc = tbt_fwu_Controller_isInSafeMode(data->controllers[i], &safe_mode);
 		if (rc != TBT_OK) {
-			g_warning ("failed to get tbd vendor ID: %s",
+			g_warning ("failed to get controller status: %s",
 				   tbt_strerror (rc));
 			continue;
 		}
 
-		/* get the model ID */
-		rc = tbt_fwu_Controller_getModelID (data->controllers[i],
-						    &info->model_id);
-		if (rc != TBT_OK) {
-			g_warning ("failed to get tbd model ID: %s",
-				   tbt_strerror (rc));
-			continue;
+		if (safe_mode)
+		{
```

Comment:
For integer checks can you make this explicit, please, like if (safe_mode != 0) and also have the brace on the same line please.